Remove dep on get_installed_distributions

LuaMilkshake opened the pull request:
This causes issues with the latest pip 18, where get_installed_distributions doesn't exist (even in _internal). It's clear pip doesn't intend for that function to be used. I've changed the call to use the suggestion on pip's issue [0], but I don't know enough about this part of Python to know if that has unintended side effects. Someone in-the-know should look at this before merging.

[0] pypa/pip#5243

jtschichold commented:
Hi @LuaMilkshake,
thanks for your PR. It looks good, the original code was making use of internal call to avoid caching when extensions were installed on the fly. I think we should accept your PR and find a better way of getting rid of those cache effects.
Please could you:
- add your name to the AUTHORS file
- retarget your PR to the `develop` branch
- explicitly state that you accept the Apache license

Thanks again

LuaMilkshake commented:
Hi Luigi,

I've rebased this over develop and added my name to AUTHORS.

I accept licensing my contributions under the Apache 2.0 license.

Thanks! Hope to see this on master soon!

jtschichold commented:
:shipit:

jtschichold commented:
@LuaMilkshake I do some tests and then cut the release (`develop` will be merged into `master`)
